server/etcdserver/server.go

```diff
@@ -294,6 +295,17 @@ type EtcdServer struct {
 	*AccessController
 }
 
+type backendHooks struct {
+	indexer cindex.ConsistentIndexer
+	lg      *zap.Logger
+}
+
+func (bh *backendHooks) OnPreCommitUnsafe(tx backend.BatchTx) {
+	if bh.indexer != nil {
```

Comment:
I prefer that we keep the error log when somehow indexer is nil, as this might result in data loss or data corruption.

Reply:
Fixed.